container-search/src/main/java/com/yahoo/prelude/query/WeakAndItem.java

```diff
@@ -87,6 +87,32 @@ protected void encodeThis(ByteBuffer buffer) {
         putString(index, buffer);
     }
 
+    private WeakAndItem foldSegments() {
+        var result = new WeakAndItem(this.index, this.n);
+        for (var child : items()) {
+            if (child instanceof SegmentItem segment && segment.shouldFoldIntoWand()) {
+                for (var subItem : segment.items()) {
+                    result.addItem(subItem);
+                }
+            } else {
+                result.addItem(child);
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public int encode(ByteBuffer buffer) {
```

Comment:
super.encode will encode this, then encode all the items in this (without folding). Won't the total result of this be a bit strange?

Reply:
super.encode() should be called when no folding is needed; maybe the logic was not clear. will refactor some...